drivers/serial/uart_pl011.c

```diff
@@ -459,132 +459,65 @@ void pl011_isr(const struct device *dev)
 }
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 
-
-#ifdef CONFIG_UART_PL011_PORT0
-
-#ifdef CONFIG_UART_INTERRUPT_DRIVEN
-static void pl011_irq_config_func_0(const struct device *dev);
-#endif
-
-static struct pl011_config pl011_cfg_port_0 = {
-	.uart = (volatile struct pl011_regs *)DT_INST_REG_ADDR(0),
-	.sys_clk_freq = DT_INST_PROP_BY_PHANDLE(0, clocks, clock_frequency),
-#ifdef CONFIG_UART_INTERRUPT_DRIVEN
-	.irq_config_func = pl011_irq_config_func_0,
-#endif
-};
-
-static struct pl011_data pl011_data_port_0 = {
-	.baud_rate = DT_INST_PROP(0, current_speed),
-};
-
-DEVICE_DT_INST_DEFINE(0,
-		    &pl011_init,
-		    NULL,
-		    &pl011_data_port_0,
-		    &pl011_cfg_port_0, PRE_KERNEL_1,
-		    CONFIG_SERIAL_INIT_PRIORITY,
-		    &pl011_driver_api);
-
-#ifdef CONFIG_UART_INTERRUPT_DRIVEN
-static void pl011_irq_config_func_0(const struct device *dev)
-{
-#if DT_NUM_IRQS(DT_INST(0, arm_pl011)) == 1
-	IRQ_CONNECT(DT_INST_IRQN(0),
-		    DT_INST_IRQ(0, priority),
-		    pl011_isr,
-		    DEVICE_DT_INST_GET(0),
-		    0);
-	irq_enable(DT_INST_IRQN(0));
-#else
-	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, tx, irq),
-		    DT_INST_IRQ_BY_NAME(0, tx, priority),
-		    pl011_isr,
-		    DEVICE_DT_INST_GET(0),
-		    0);
-	irq_enable(DT_INST_IRQ_BY_NAME(0, tx, irq));
-
-	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, rx, irq),
-		    DT_INST_IRQ_BY_NAME(0, rx, priority),
-		    pl011_isr,
-		    DEVICE_DT_INST_GET(0),
-		    0);
-	irq_enable(DT_INST_IRQ_BY_NAME(0, rx, irq));
-
-	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, rxtim, irq),
-		    DT_INST_IRQ_BY_NAME(0, rxtim, priority),
-		    pl011_isr,
-		    DEVICE_DT_INST_GET(0),
-		    0);
-	irq_enable(DT_INST_IRQ_BY_NAME(0, rxtim, irq));
-#endif
-}
-#endif
-
-#endif /* CONFIG_UART_PL011_PORT0 */
-
-#ifdef CONFIG_UART_PL011_PORT1
-
-#ifdef CONFIG_UART_INTERRUPT_DRIVEN
-static void pl011_irq_config_func_1(const struct device *dev);
-#endif
-
-static struct pl011_config pl011_cfg_port_1 = {
-	.uart = (volatile struct pl011_regs *)DT_INST_REG_ADDR(1),
-	.sys_clk_freq = DT_INST_PROP_BY_PHANDLE(1, clocks, clock_frequency),
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
-	.irq_config_func = pl011_irq_config_func_1,
-#endif
-};
-
-static struct pl011_data pl011_data_port_1 = {
-	.baud_rate = DT_INST_PROP(1, current_speed),
-};
-
-DEVICE_DT_INST_DEFINE(1,
-		    &pl011_init,
-		    NULL,
-		    &pl011_data_port_1,
-		    &pl011_cfg_port_1, PRE_KERNEL_1,
-		    CONFIG_SERIAL_INIT_PRIORITY,
-		    &pl011_driver_api);
+#define PL011_IRQ_CONFIG_FUNC_BODY(n, prop, i) \
+	{ \
+		IRQ_CONNECT(DT_IRQ_BY_IDX(n, i, irq), \
+			DT_IRQ_BY_IDX(n, i, priority), \
+			pl011_isr, \
+			DEVICE_DT_GET(n), \
+			0); \
+		irq_enable(DT_IRQ_BY_IDX(n, i, irq)); \
+	}
```

Comment:
nit: please use a common format to align the `\` at the end of the line.

Reply:
Done